fhir-persistence-cassandra/src/main/java/com/ibm/fhir/persistence/cassandra/cql/DatasourceSessions.java

```diff
@@ -0,0 +1,210 @@
+            }
+            
+            CassandraPropertyGroupAdapter adapter = new CassandraPropertyGroupAdapter(connectionProps);
+            return getDatabaseSession(key, adapter, true);
+        }
+        catch (Exception x) {
+            throw new IllegalStateException(x);
+        }
+    }
+    
+    /**
+     * Get the CqlSession for the Cassandra database matching the configuration defined
+     * by the properties adapter.
+     * @param adapter
+     * @return
+     */
+    private static CqlSession getDatabaseSession(TenantDatasourceKey key, CassandraPropertyGroupAdapter adapter, boolean setKeyspace) {
+        CqlSessionBuilder builder = CqlSession.builder();
+        
+        for (ContactPoint cp: adapter.getContactPoints()) {
+            builder.addContactPoint(new InetSocketAddress(cp.getHost(), cp.getPort()));
+        }
+        builder.withLocalDatacenter(adapter.getLocalDatacenter());
+
+        if (setKeyspace) {
+            // Use the tenant id value directly as for the keyspace
+            builder.withKeyspace(key.getTenantId());
+        }
+        
+        return builder.build();
+    }
+
+    /**
+     * Create a special session without specifying the keyspace, which is needed to support
+     * schema creation where we create the keyspace for the first time
+     * @return
+     */
+    public static CqlSession getSessionForBootstrap(String tenantId, String dsId) {
+
+        String dsPropertyName = FHIRConfiguration.PROPERTY_DATASOURCES + "/" + dsId;
+        PropertyGroup dsPG = FHIRConfigHelper.getPropertyGroup(dsPropertyName);
+        if (dsPG == null) {
+            throw new IllegalStateException("Could not locate configuration property group: " + dsPropertyName);
+        }
+
+        try {
+            // Get the datasource type (should be cassandra in this case).
+            String type = dsPG.getStringProperty("type", null);
+            if (type == null) {
+                throw new IllegalStateException("Could not locate 'type' property within datasource property group: " + dsPropertyName);
+            }
+    
+            // Confirm that this is a cassandra datasource configuration element
+            if (!"cassandra".equals(type)) {
+                throw new IllegalStateException("Unsupported 'type' property value within datasource property group: " + type);  
+            }
+    
+            // Get the connection properties
+            PropertyGroup connectionProps = dsPG.getPropertyGroup("connectionProperties");
+            if (connectionProps == null) {
+                throw new IllegalStateException("Could not locate 'connectionProperties' property group within datasource property group: " + dsPropertyName);
+            }
+            
+            TenantDatasourceKey key = new TenantDatasourceKey(tenantId, dsId);
+            CassandraPropertyGroupAdapter adapter = new CassandraPropertyGroupAdapter(connectionProps);
+            return getDatabaseSession(key, adapter, false);
```

Comment:
A lot of common code here and in newSession() - consider moving to private method